chain/ethereum/src/adapter.rs

```diff
@@ -173,6 +216,33 @@ pub(crate) struct EthereumLogFilter {
     wildcard_events: HashSet<EventSignature>,
 }
 
+impl Into<Vec<LogFilter>> for EthereumLogFilter {
+    fn into(self) -> Vec<LogFilter> {
+        self.contracts_and_events_graph
```

Comment:
This is not accounting for `wildcard_events`. For non-wildcards, it's generating going to effectively generate all relevant `(contract, log)` pairs since it adds one filter per edge. This is not optimal when there are many contracts associated to log or vice versa.

You solve all that by just calling `eth_get_logs_filters` here and trivially converting the output to `LogFilter`. That algorithm is probably overkill with static filters, but since we already have it why not use it.